### PR TITLE
Added a popgun generator example into popgun repo. Can be viewed at localhost:8080/perf.html

### PR DESCRIPTION
**Pull request by matthewHuiCsc**
Note that I had to tweak iqb a bit to include the perf.html file. If any objections, let me know!

### CODE REVIEW

File: app/perf/example-generator.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+export default class ExampleGenerator {
+
+	constructor() {
+		this.colorGenerator = new ColorGenerator();
+		this.id = 0;
+	}
+
+	public createPopgun(): Element {
+		let trigger = document.createElement('div');
```
Comment (danyaal) [inline app/perf/example-generator.ts]: This is more accurate to call `target` than `trigger`.

File: app/perf.html [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  <head>
+    <meta charset="UTF-8">
+    <title>popgun demo</title>
+    <link rel="stylesheet" type="text/css" href="app.css">
+
+    <!-- merchants -->
+    <link href='https://fonts.googleapis.com/css?family=Lato:400,400italic,700,700italic,900,900italic,300italic,300,100italic,100' rel='stylesheet' type='text/css'>
+    <style type='text/css' href='app.css'></style>
+    <style>
+    .target {
```
Comment (danyaal) [inline app/perf.html]: I don't think you use most of these classes, so feel free to remove w/e you don't need.

**Comment by danyaal** [comment]
Can we also make sure we `.npmignore` anything that doesn't need to be published?

File: app/perf.html (hunk 2 above)
Comment (matthewHuiCsc) [inline app/perf.html]: Ah fair point, will remove the `<link href="googleapis-thing">` and `<style>` elements.

**Comment by danyaal** [comment]
Besides my minor comments and the `.npmignore` this LGTM! 👍

**Comment by matthewHuiCsc** [comment]
Hm? All the files there are needed though.

File: app/perf/example-generator.ts (hunk 1 above)
Comment (matthewHuiCsc) [inline app/perf/example-generator.ts]: Sounds good!

**Comment by danyaal** [comment]
But `.npmignore` will just ignore things that don't need to be published to the npm repo. All we need for that is the compiled js files.

**Pull request merged**